torch/csrc/utils/tensor_new.cpp

```diff
@@ -216,8 +216,10 @@ Tensor internal_new_from_data(const Type & type, at::optional<Device> device_opt
                                                        *torch::getLayout(type.backend()),
                                                        type_inference_device_type);
       const auto& type_to_use = type_inference ? type_inference_type : type;
-      return copy_variables ? new_with_tensor_copy(type_to_use, var, device_index) :
+      auto new_tensor = copy_variables ? new_with_tensor_copy(type_to_use, var, device_index) :
                               new_with_type_conversion(type_to_use, var, device_index);
+      if (!args_requires_grad) new_tensor.detach_();
```

Comment:
No, this still looks wrong. Based on the proposed semantics in my comment, you should *unconditionally* detach, and then depending on `args_require_grad` set the `requires_grad` back to True (turning it into a requires_grad leaf variable).